src/views/ApartmentInfo.vue

```diff
@@ -81,50 +190,234 @@ export default defineComponent({
+    });
+
+    const setOwnerships = async () => {
+      await store.dispatch(
+        `${StoreModuleEnum.ownershipsStore}/${OwnershipsActionEnum.SET_OWNERSHIPS}`,
+        apartment.value
+      );
+      loading.value = false;
+    };
+
+    const ownershipsData = computed(() => {
+      return store.getters[`${StoreModuleEnum.ownershipsStore}/${OwnershipsGettersEnum.getOwnershipsData}`];
+    });
+
+    const initDataTable = () => {
+      ownershipsInfo.value = [];
+      ownershipsData.value.map((el: any) => {
+        const currentContact = el?.owner.contacts.find(
+          (contact: UserContactInterface) => !!contact.main && contact.type === ContactTypeEnum.email
+        );
+
+        const newElem = {
+          id: el?.id,
+          fullName: `${el?.owner.firstName} ${el?.owner.middleName} ${el?.owner.lastName}`,
+          contact: currentContact.email,
+          ownershipPart: el?.ownershipPart,
+          voutsPart: 0,
+        };
+
+        ownershipsInfo.value.push(newElem);
+      });
+    };
+
+    const calcOwnerVoutingShare = computed(() => {
+      if (loading.value) {
+        return [];
+      } else {
+        return ownershipsData.value.reduce((acc: any, cur: OwnershipsModel) => {
+          const ownershipsPartNumber = convertFractionToDecimal(cur.ownershipPart);
+          const voutingShare =
+            (ownershipsPartNumber * (apartmentInfo?.value?.apartmentArea || 1) * 100) /
+            (houseInfo?.value?.houseArea || 1);
+          const element = {
+            id: cur.id,
+            ownershipPart: voutingShare.toFixed(3),
+          };
+          acc.push(element);
+          return acc;
+        }, []);
+      }
+    });
+
+    const deleteOwner = () => {
+      const payload = {
+        apartmentId: apartment.value,
+        ownerId: selectedOwner.value.id,
+      };
+      store.dispatch(`${StoreModuleEnum.ownershipsStore}/${OwnershipsActionEnum.DELETE_OWNER}`, payload);
+      deleteOwnerDialog.value = false;
+    };
+
+    const editOwner = () => {
+      if (editOwnershipData.ownershipPart === selectedOwner.value.ownershipPart) {
+        editOwnerDialog.value = false;
+        return;
+      }
+      const payload = {
+        apartmentId: apartment.value,
+        ownerId: selectedOwner.value.id,
+        data: new UpdateOwnershipsDTOModel({
+          ownershipPart: selectedOwner.value.ownershipPart,
+        }),
+      };
+      store.dispatch(`${StoreModuleEnum.ownershipsStore}/${OwnershipsActionEnum.EDIT_OWNER}`, payload);
+      editOwnerDialog.value = false;
+    };
+
+    const convertFractionToDecimal = (fraction: string): number => {
+      if (fraction === '1') {
+        return 1;
+      } else if (fraction === '0') {
+        return 0;
+      }
+      const splitFraction = fraction.split('/');
+      const decimal = (+splitFraction[0] / +splitFraction[1]).toFixed(4);
+      return Number(decimal);
+    };
```

Comment:
`
  const splitFraction = fraction.split('/');
  
  if (splitFraction.length === 1) return parseInt(fraction);
  
  const decimal = (+splitFraction[0] / +splitFraction[1]).toFixed(4);
  return Number(decimal);
`

Reply:
splitFraction.length cann't be equal 1 cause then we will have an error previously on line 361 when we try to split it. 
I think you mean the checking on line 356, so I can replace it by one line:
 if (fraction.length === 1) return parseInt(fraction);
but before splitting.
And it's a good idea, thanks!